pallets/creditcoin/src/tests.rs

```diff
@@ -3082,13 +3107,13 @@ fn register_address_v2_should_error_when_address_too_long() {
 		let proof = OwnershipProof::EthSign(ownership_proof);
 		assert_noop!(
 			Creditcoin::register_address_v2(Origin::signed(who), blockchain, address, proof),
-			crate::Error::<Test>::EthSignExternalAddressGenerationFailed
+			crate::Error::<Test>::MalformedExternalAddress
```

Comment:
+1 on this, now just needs adjustments to `try_extract_address()` so that it doesn't short-circuit the execution and allows `register_address_v2` to reach this snippet 

```
                                        // note: this error condition is unreachable!
                                        // AddressFormatNotSupported or OwnershipNotSatisfied will error out first
                                        ensure!(
                                                helpers::address_is_well_formed(&blockchain, &address),
                                                Error::<T>::MalformedExternalAddress
                                        );
```

See for example: https://blog.logrocket.com/mocking-rust-mockall-alternatives/ but I don't think we use real mocking (in the sense of replacing dependency functions) anywhere. 

Note: looking at the comments in the source code though I see this commit: beb52bb9a458df806aa8677492681621d5bf8f41. IIRC at the time we decided to let it slide b/c we had other priorities rather than trying to figure out how to properly mock things.

If you want you can give it a try. Alternatively we can remove this test scenario together with the original `register_address_v2_should_error_when_signature_is_invalid` and replace them with new test scenarios which target the underlying helper functions directly. 

Let me know if you want me to jump on this and give it a try myself.

Reply:
This is also addressed by moving the errors that I mentioned above. I think the the tests better reflect the behaviour we are looking for without mocking anything but I think its probably a good idea to start gaining experience in that arena as soon as possible. Im going to start looking into mocking alternatives